libgalois/include/katana/PropertyGraph.h

```diff
@@ -473,80 +495,172 @@ class KATANA_EXPORT PropertyGraph {
     return edge_properties()->schema();
   }
 
-  /// \returns the number of node types
-  size_t GetNodeTypesNum() const {
-    return node_type_name_to_type_set_ids_.size();
+  /// \returns the number of node atomic types
+  size_t GetNumNodeAtomicTypes() const {
+    return node_atomic_type_name_to_entity_type_id_.size();
   }
 
-  /// \returns the number of edge types
-  size_t GetEdgeTypesNum() const {
-    return edge_type_name_to_type_set_ids_.size();
+  /// \returns the number of edge atomic types
+  size_t GetNumEdgeAtomicTypes() const {
+    return edge_atomic_type_name_to_entity_type_id_.size();
   }
 
-  /// \returns true if a node type with @param name exists
-  /// NB: no node may have this type
+  /// \returns the number of node entity types (including kUnknownType)
+  size_t GetNumNodeEntityTypes() const {
+    return node_entity_type_id_to_atomic_entity_type_ids_.size();
+  }
+
+  /// \returns the number of edge entity types (including kUnknownType)
+  size_t GetNumEdgeEntityTypes() const {
+    return edge_entity_type_id_to_atomic_entity_type_ids_.size();
+  }
+
+  /// \returns true iff a node atomic type @param name exists
+  /// NB: no node may have a type that intersects with this atomic type
```

Comment:
You could just say "no node may have this atomic type". It means the same thing.